rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala

```diff
@@ -291,7 +291,14 @@ class PrepareTemplateVariablesImpl(
     }
 
     //utilitary method for formating an input list
-    def formatInputs(x: Seq[(Technique, String)]) = x.map(_._2).distinct.mkString("\"", s"""",\n${" "*14}"""", s""""\n  """)
+    def formatInputs(x: Seq[(Technique, String)]) = {
+      val inputs = x.map(_._2).distinct
+      if (inputs.isEmpty) {
+        ""
+      } else {
+        x.map(_._2).distinct.mkString("\"", s"""",\n${" "*14}"""", s""""\n  """)
```

Comment:
You can replace 'x.map(_._2).distinct' by 'inputs' since it is already computed :)